Pathogens/Utility/location_utils/lat_lng_generation_report.py

```diff
@@ -0,0 +1,314 @@
+    city_name: str | None,\
+    state_name: str | None,\
+    country_name: str,\
+    logLevel: MapboxApiRequestLogLevel
+):
+    string_log_level_printout_map = {
+        MapboxApiRequestLogLevel.INFO: "[INFO]",
+        MapboxApiRequestLogLevel.WARN: "[WARN]",
+        MapboxApiRequestLogLevel.ERROR: "[ERROR]",
+    }
+    city_state_country_printout = "[" + (city_name if city_name is not None else 'N/A') + ", " + \
+        (state_name if state_name is not None else 'N/A') + ", " + country_name + "]"
+
+
+    return string_log_level_printout_map[logLevel] + " - " + city_state_country_printout
+
+def format_mapbox_text_and_matching_text_for_display(mapbox_response):
+    if(mapbox_response is None):
+        return "N/A"
+    elif(mapbox_response.text is not None and mapbox_response.matching_text is not None):
+        return '"' + mapbox_response.text + '"/"' + mapbox_response.matching_text + '"'
+    elif(mapbox_response.text is not None and mapbox_response.matching_text is None):
+        return '"' + mapbox_response.text + '"'
+    elif(mapbox_response.text is None and mapbox_response.matching_text is not None):
+        return '"' + mapbox_response.matching_text + '"'
+    else:
+        return "N/A"
+
+def format_mapbox_response_as_string(mapbox_response):
+    if(mapbox_response is None):
+        return "[NO RESPONSE]"
+    
+    bounding_box_as_string = "[" + ', '.join(map(str, mapbox_response.bounding_box)) + "]"\
+        if mapbox_response.bounding_box is not None else "N/A"
+    center_coordinates_as_string = "[" + ', '.join(map(str, mapbox_response.center_coordinates)) + "]"\
+        if mapbox_response.bounding_box is not None else "N/A"
+    text_as_string = mapbox_response.text if mapbox_response.text is not None else "N/A"
+    matching_text_as_string = mapbox_response.matching_text if mapbox_response.matching_text is not None else "N/A"
+    region_name_as_string = mapbox_response.region_name if mapbox_response.region_name is not None else "N/A"
+
+    return "[" + "center_coordinates: " + center_coordinates_as_string \
+        + ", bounding_box: " + bounding_box_as_string \
+        + ", text: " + text_as_string \
+        + ", matching_text: " + matching_text_as_string \
+        + ", region_name: " + region_name_as_string + "]"
+
+def generate_line_for_response_and_request(\
+    city_name: str | None,\
+    state_name: str | None,\
+    country_name: str,\
+    mapbox_request_url: str,\
+    mapbox_response\
+):
+    prefix = generate_mapbox_api_request_log_prefix(
+        logLevel = MapboxApiRequestLogLevel.INFO,
+        city_name = city_name,
+        state_name = state_name,
+        country_name = country_name
+    )
+    request_with_api_key_redacted = "[" + redact_mapbox_api_key_from_mapbox_api_query_url(mapbox_request_url) + "]"
+
+    return prefix + " - " + request_with_api_key_redacted + " - " + format_mapbox_response_as_string(mapbox_response)
+
+def generate_line_for_text_consitency_check(
```

Comment:
Context: I want to log a warning every single time the city name we have and the city name mapbox gives are different.

This:
- Caught a lot of typos in our city names (Ex. we have a `Jakarata` instead of `Jakarta` and a `Montsinnery–Tonnegrande` instead of `Montsinéry-Tonnegrande`). It's not perfect and mapbox actually has some typos/questionable names as well but it's at least a flag for the person reading the report to double-check the spelling of the name.
- Is sometimes indicative of a larger issue like mapbox picking the wrong city.